session/session.go

```diff
@@ -664,11 +664,8 @@ func (s *session) retry(ctx context.Context, maxCnt uint) (err error) {
 			if err != nil {
 				return err
 			}
-			pessTxnConf := config.GetGlobalConfig().PessimisticTxn
-			if pessTxnConf.Enable {
-				if s.sessionVars.TxnMode == ast.Pessimistic {
-					s.sessionVars.TxnCtx.IsPessimistic = true
-				}
+			if !s.sessionVars.IsAutocommit() {
+				s.sessionVars.SetStatusFlag(mysql.ServerStatusInTrans, true)
```

Comment:
I think it should be set even if `autocommit` is on. For example, if the transaction is committed by `set autocommit = 1`, the `mysql.ServerStatusInTrans` flag won't be set and the point-get executor still reads data from the store's snapshot.

Reply:
Thanks for the reminding, I'll add test cases to cover such path.